### DISCUSSION TITLE
IRIS security integration

### DISCUSSION THREAD

**Discussion by rstaylor** [intersystems-community/vscode-objectscript#1296] [Q&A] [answered]
What security features of IRIS does the terminal take advantage of?

Is there any way to limit the actions a user can perform at the terminal prompt?

The concern here is providing only limited access to a terminal prompt for running or testing client written code in a managed cloud environment.

**Comment by isc-bsaviano** [comment, marked as answer]
@rstaylor For the WebSocket terminal, the user needs to have %Development to use it (same as the rest of the Atelier API). Once you're in, basic ObjectScript-level security is all you have. For example, if you try to `zn` into a namespace you don't have the resource for, you'll get a `<PROTECT>`. No checks beyond that are performed.
